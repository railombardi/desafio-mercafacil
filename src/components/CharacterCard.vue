<template>
  <div class="card">
    <img class="card__img" :src="image" />
    <div class="card__info">
      <span class="name"> {{ name }}</span>
      <span class="species"> {{ species }} </span>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    name: {
      type: String,
      default: "",
    },
    image: {
      type: String,
      default: "",
    },
    species: {
      type: String,
      default: "",
    },
  },
};
</script>

<style lang="scss" scoped>
.card {
  display: grid;
  grid-template-rows: 168px 76px;
  width: 240px;
  box-shadow: 0px 1px 5px rgba(0, 0, 0, 0.2), 0px 3px 4px rgba(0, 0, 0, 0.12),
    0px 2px 4px rgba(0, 0, 0, 0.14);
  border-radius: 4px;
  transition: all 0.3s ease;
  cursor: pointer;
  &:hover {
    width: 260px;
    transform: translateX(-10px);
  }
  &__img {
    width: 100%;
    height: 100%;
    object-fit: cover;
    border-radius: 4px 4px 0;
  }
  &__info {
    display: flex;
    flex-direction: column;
    max-width: 240px;
    padding: 12px 16px;
    .name {
      font-weight: 500;
      font-size: 20px;
      line-height: 30px;
      max-width: 100%;
      white-space: nowrap;
      text-overflow: ellipsis;
      overflow: hidden;
    }
    .species {
      font-size: 14px;
      line-height: 21px;
      color: rgba(0, 0, 0, 0.6);
    }
  }
}
</style>
