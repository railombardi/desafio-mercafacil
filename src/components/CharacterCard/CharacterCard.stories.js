import CharacterCard from "./CharacterCard.vue";
export default {
  title: "Components/CharacterCard",
  component: CharacterCard,
  argTypes: {
    species: {
      control: { type: "select" },
      options: ["human", "alien", "robot"],
      defaultValue: "human",
    },
  },
};

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: {
    CharacterCard,
  },
  template: `<div style="display: flex; justify-content: center">
              <character-card :name="name" :image="image" :species="species" />
             </div>`,
});

export const Default = Template.bind({});
Default.args = {
  name: "Rick Sanch√©z",
  image: "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
};
export const OtherCharacter = Template.bind({});
OtherCharacter.args = {
  name: "Morty Smith",
  image: "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
};
