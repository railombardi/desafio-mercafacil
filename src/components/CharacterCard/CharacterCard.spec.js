import { shallowMount, createLocalVue } from "@vue/test-utils";
import CharacterCard from "./CharacterCard.vue";

const localVue = createLocalVue();

let wrapper = shallowMount(CharacterCard, {
  localVue,
});

describe("CharacterCard.vue", () => {
  it("Snapshot test", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  describe("[PROPS] =>", () => {
    test("When the prop image receives value, the content is reproduced by <img> component", () => {
      wrapper.setProps({
        image: "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
      });
      const img = wrapper.findAll(".card__img");
      expect(img.length).toBe(1);
    });
    it("When the prop name receives a value, the content is displayed in the target span", async () => {
      const text = "Test name prop";
      await wrapper.setProps({ name: text });
      const span = wrapper.find(".name");
      expect(span.text()).toBe(text);
    });
    it("When the prop species receives a value, the content is displayed in the target span", async () => {
      const text = "Test species prop";
      await wrapper.setProps({ species: text });
      const span = wrapper.find(".species");
      expect(span.text()).toBe(text);
    });
  });
});
