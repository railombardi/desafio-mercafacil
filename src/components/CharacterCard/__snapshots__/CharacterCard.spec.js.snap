// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard.vue Snapshot test 1`] = `
<div class="card"><img class="card__img">
  <div class="card__info"><span class="name"> </span> <span class="species">  </span></div>
</div>
`;
